plugins/core/gs-plugin-appstream.c

```diff
@@ -91,6 +91,36 @@ gs_plugin_appstream_convert_component_kind (const gchar *kind)
 	return kind;
 }
 
+static gboolean
+gs_plugin_appstream_override_app_id_cb (XbBuilderFixup *self,
+                                        XbBuilderNode *bn,
+                                        gpointer user_data,
+                                        GError **error)
+{
+	if (g_strcmp0 (xb_builder_node_get_element (bn), "component") == 0) {
+		g_autoptr(XbBuilderNode) id = xb_builder_node_get_child (bn, "id", NULL);
+		g_autoptr(XbBuilderNode) launchable = xb_builder_node_get_child (bn, "launchable", NULL);
+
+		if (launchable != NULL && id != NULL) {
+			const gchar *type = xb_builder_node_get_attr (launchable, "type");
+
+			if (g_strcmp0 (type, "desktop-id") == 0) {
+				const gchar *app_id = xb_builder_node_get_text (id);
+				const gchar *launchable_id = xb_builder_node_get_text (launchable);
+
+				if (app_id != NULL &&
+				    launchable_id != NULL &&
+				    g_strcmp0 (app_id, launchable_id) != 0) {
+					g_debug ("Overriding appdata app-id %s with <launchable> desktop-id: %s",
```

Comment:
I don't quite understand the commit message here. Why would this be needed in Endless OS and not upstream? Mysterious.